router/src/server.rs

```diff
@@ -374,19 +344,93 @@ responses(
 example = json ! ({"error": "unhealthy", "error_type": "healthcheck"})),
 )
 )]
-#[instrument(skip(health))]
 /// Health check method
-async fn health(mut health: Extension<Health>) -> Result<(), (StatusCode, Json<ErrorResponse>)> {
-    match health.check().await {
-        true => Ok(()),
-        false => Err((
-            StatusCode::SERVICE_UNAVAILABLE,
-            Json(ErrorResponse {
-                error: "unhealthy".to_string(),
-                error_type: "healthcheck".to_string(),
-            }),
-        )),
+async fn health(
```

Comment:
I wonder if we can just have this function return true if the metrics indicate the server has successfully completed a request in the last N seconds. I worry in high throughput scenarios that hitting this repeatedly could slow things down. But maybe we can leave that as a TODO.


cc @noyoshi

Reply:
This was how the old endpoint was implemented. We can probably just store a bool and do lightweight checks. See: https://github.com/predibase/lorax/blob/main/router/src/health.rs#L25-L75

Reply:
I like the use of the `generation_health` var, but would want to timestamp it so we re-run after some amount of time.

Fine to merge it as-is for now, but would like to revisit this soon as I do feel it's going to slow things down in the current form.